File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        out = [1] * len(nums)
        print(out)
        p = 1
        for i, n in enumerate(nums):
            print("j", i)
            out[i] = p 
            p *= n
        p = 1
        for i in range(len(nums)-1,-1, -1):
            out[i] *= p
            p *= nums[i]
        return out

File: Data Structures & Algorithms/find-first-and-last-position-of-element-in-sorted-array/submission-0.ts
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    searchRange(nums: number[], target: number): number[] {
        const start = nums.findIndex(x => x === target);
        if (start === -1) return [-1, -1]
        let end = start + 1
        while (end < nums.length && nums[end] === target) {
            end += 1
        }
        return [start, end -1]
    }
}
